ops/testing.py

```diff
@@ -1253,14 +1259,17 @@ def storage_get(self, storage_name_id, attribute):
                 'ERROR invalid value "{}/{}" for option -s: storage not found'.format(name, index))
 
     def storage_add(self, name: str, count: int = 1):
+        if name.startswith('/'):
+            raise model.ModelError('test storage name cannot start with "/"')
+
         if name not in self._storage_list:
             self._storage_list[name] = {}
         result = []
         for i in range(count):
             index = self._storage_id_counter
             self._storage_id_counter += 1
             self._storage_list[name][index] = {
-                "location": "/{}/{}".format(name, index)
+                "location": os.path.join(self._harness_tmp_dir.name, name, str(index)),
```

Comment:
Nit: we already had `pathlib` imported in this file, and iirc it's used fairly extensively elsewhere in the codebase to join paths together?

```suggestion
                "location": pathlib.Path(self._harness_tmp_dir.name) / name / str(index),
```

Reply:
There's other code that would need to be changed to accommodate this; also only changing this attrib would leave logic around this more bifurcated on path handling.  I think this is probably better left for another more filepath focused PR.

Reply:
https://github.com/canonical/operator/issues/742

Reply:
Ack, thanks.